src/dbnode/storage/bootstrap/bootstrapper/peers/source.go

```diff
@@ -642,73 +648,123 @@ func (s *peersSource) readIndex(
 		zap.Int("concurrency", concurrency),
 	)
 
+	groupFn := bootstrapper.NewShardTimeRangesTimeWindowGroups
+	groupedByBlockSize := groupFn(shardsTimeRanges, dataBlockSize)
```

Comment:
Hm, should this be index block size instead of data block size? Just comparing to the `fs/source.go` implementation.

Reply:
Hard to know, I see we did the data block size for fetching below.